tools/lxdclient/client_image.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// +build go1.3
+
+package lxdclient
+
+import (
+	"os/exec"
+
+	"github.com/lxc/lxd/shared"
+
+	"github.com/juju/errors"
+)
+
+type rawImageClient interface {
+	ListAliases() ([]shared.ImageAlias, error)
+}
+
+type imageClient struct {
+	raw rawImageClient
+}
+
+func (i imageClient) EnsureImageExists(series string) error {
+	name := i.ImageNameForSeries(series)
```

Comment:
do we have to worry about architecture? Or that is naturally determined just by the local host architecture?